cdap-common/bin/functions.sh

```diff
@@ -493,11 +491,34 @@ cdap_set_spark() {
         return 1
       fi
       SPARK_HOME=$(echo -e "${SPARK_VAR_OUT}" | grep ^SPARK_HOME= | cut -d= -f2)
-      export SPARK_HOME
-      return 0
     fi
-    return 1
   fi
+
+  export SPARK_HOME
+
+  # Find environment variables setup via spark-env.sh
+  SPARK_ENV_SH=${SPARK_HOME}/conf/spark-env.sh
+  if [[ -f ${SPARK_ENV_SH} ]]; then
+    if [[ -z ${SPARK_ENV_PATTERN} ]]; then
+      # By default, only get env from spark-env.sh that starts with PY or SPARK
+      SPARK_ENV_PATTERN="^(PY|SPARK)"
+    fi
+    SPARK_ENV=$(source ${SPARK_ENV_SH}; env | grep -E "${SPARK_ENV_PATTERN}")
+    __ret=$?
+    if [[ ${__ret} -eq 0 ]]; then
+      local readonly __env_out=$(mktemp)
+      echo "${SPARK_ENV}" | while read -r line; do
+        # Prefix the env variable with _SPARK_ to avoid conflicts
+        echo "export _SPARK_${line%%=*}=${line#*=}" >> ${__env_out}
+      done
+      source ${__env_out}
+      rm ${__env_out}
```

Comment:
The prefixing is for CDAP master to grab the set of spark env that needs to pass to the container shell.